### PR TITLE
Refactor react query keys

### PR DESCRIPTION
**Pull request by biskuvit**
Test me! 'MASTER'
[Link to REFACTOR-REACT-QUERY-KEYS](https://refacto-refine.pankod.com)

Hey, I'm client! Test me! 'MASTER'
[Link to REFACTOR-REACT-QUERY-KEYS](https://refacto-client.refine.pankod.com)

Test me! 'MASTER'
[Link to REFACTOR-REACT-QUERY-KEYS](https://refacto-refine.pankod.com)

Hey, I'm client! Test me! 'MASTER'
[Link to REFACTOR-REACT-QUERY-KEYS](https://refacto-client.refine.pankod.com)

Test me! 'MASTER'
[Link to REFACTOR-REACT-QUERY-KEYS](https://refacto-refine.pankod.com)

Hey, I'm client! Test me! 'MASTER'
[Link to REFACTOR-REACT-QUERY-KEYS](https://refacto-client.refine.pankod.com)

Test me! 'MASTER'
[Link to REFACTOR-REACT-QUERY-KEYS](https://refacto-refine.pankod.com)

Hey, I'm client! Test me! 'MASTER'
[Link to REFACTOR-REACT-QUERY-KEYS](https://refacto-client.refine.pankod.com)

Test me! 'MASTER'
[Link to REFACTOR-REACT-QUERY-KEYS](https://refacto-refine.pankod.com)

Hey, I'm client! Test me! 'MASTER'
[Link to REFACTOR-REACT-QUERY-KEYS](https://refacto-client.refine.pankod.com)

Test me! 'MASTER'
[Link to REFACTOR-REACT-QUERY-KEYS](https://refacto-refine.pankod.com)

Hey, I'm client! Test me! 'MASTER'
[Link to REFACTOR-REACT-QUERY-KEYS](https://refacto-client.refine.pankod.com)

Test me! 'MASTER'
[Link to REFACTOR-REACT-QUERY-KEYS](https://refacto-refine.pankod.com)

**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

*Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request.*

Please provide enough information so that others can review your pull request:

Explain the **details** for making this change. What existing problem does the pull request solve?

**Test plan (required)**

Demonstrate the code is solid. If not, please add `WIP:` in its title.

**Closing issues**

Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: packages/core/src/hooks/data/useDelete.ts [hunk 1]
```diff
@@ -141,51 +137,44 @@ export const useDelete = <
...
+                                    );
+
+                                    return {
+                                        data,
+                                        total: previous.total - 1,
+                                    };
+                                },
+                            );
+                        } else {
+                            queryClient.invalidateQueries([resource]);
```
Comment (omeraplak) [inline packages/core/src/hooks/data/useDelete.ts:177]:
```suggestion
                            queryClient.removeQueries([resource]);
```

**Review by omeraplak — CHANGES_REQUESTED** [review]

File: packages/core/src/hooks/data/useUpdate.ts [hunk 2]
```diff
@@ -149,60 +146,97 @@ export const useUpdate = <
...
+                                        return null;
+                                    }
+                                    const data = previous.data.map(
+                                        (record: TData) => {
+                                            if (
+                                                record.id?.toString() ===
+                                                id.toString()
+                                            ) {
+                                                return {
+                                                    id: Number(id),
```
Comment (omeraplak) [inline packages/core/src/hooks/data/useUpdate.ts:180]:
```suggestion
                                                    id: Number(id),
```
why do we convert id to number? id can be a string

**Review by omeraplak — CHANGES_REQUESTED** [review]

File: packages/core/src/hooks/data/useDeleteMany.ts [hunk 3]
```diff
@@ -142,53 +139,43 @@ export const useDeleteMany = <
...
-                                queryClient.removeQueries(queryKey);
-                            }
+                                    return {
+                                        data,
+                                        total: previous.total - 1,
+                                    };
+                                },
+                            );
                         }
+                        queryClient.removeQueries([resource, "detail", ids]);
```
Comment (omeraplak) [inline packages/core/src/hooks/data/useDeleteMany.ts:177]: Are we sure this is working?

Comment (ozkalai) [reply packages/core/src/hooks/data/useDeleteMany.ts:177]: I think we forgot to delete this point, you are right, we'll remove it. Thanks for your comment 🤸‍♀️

File: packages/core/src/hooks/data/useDelete.ts [hunk 4]
```diff
@@ -225,15 +219,27 @@ export const useDelete = <
...
+                    resource,
+                    "getMany",
+                ]);
+
+                if (getManyQueries.length > 0) {
+                    queryClient.removeQueries(getManyQueries);
+                }
+
+                if (detailQueries.length > 0) {
+                    queryClient.removeQueries(detailQueries);
```
Comment (omeraplak) [inline packages/core/src/hooks/data/useDelete.ts:238]:
```suggestion
                queryClient.removeQueries([resource, "detail"]);
                queryClient.removeQueries([resource, "getMany"]);
```

**Review by omeraplak — APPROVED** [review]

**Pull request merged**